packages/mask/src/plugins/NextID/components/NextIdPage.tsx

```diff
@@ -56,22 +78,26 @@ export function NextIdPage({}: NextIDPageProps) {
     }, [personaConnectStatus, t])
 
     const { value: currentPersona, loading: loadingPersona } = useAsyncRetry(() => {
-        if (!currentProfileIdentifier) return Promise.resolve(undefined)
-        return Services.Identity.queryPersonaByProfile(currentProfileIdentifier.identifier)
-    }, [currentProfileIdentifier, personaConnectStatus.hasPersona])
+        if (!visitingPersonaIdentifier) return Promise.resolve(undefined)
+        return Services.Identity.queryPersonaByProfile(visitingPersonaIdentifier.identifier)
+    }, [visitingPersonaIdentifier, personaConnectStatus.hasPersona])
+
+    const { value: isAccountVerified, loading: loadingVerifyInfo } = useAsync(() => {
+        if (!currentPersona?.publicHexKey) return Promise.resolve(undefined)
+        return queryIsBound(currentPersona.publicHexKey, platform, visitingPersonaIdentifier.identifier.userId)
+    }, [isOwn, currentPersona, visitingPersonaIdentifier, isVerified])
 
     const { value: bindings, loading } = useAsync(async () => {
         if (!currentPersona) return
-        if (isOwn) {
-            return Services.Helper.queryExistedBinding(currentPersona.identifier)
-        }
-
-        if (!visitingPersonaIdentifier) return null
-        const visitingPersona = await Services.Identity.queryPersonaByProfile(visitingPersonaIdentifier.identifier)
+        return queryExistedBindingByPersona(currentPersona.publicHexKey!)
+    }, [currentPersona, isOwn, count])
```

Comment:
You don't need to update a count to trigger refetching, use retry from `useAsyncRetry()` instead

```tsx
const { retry: retryQueryBiding } = useAsyncRetry()
<BindingDialog onBound={retryQueryBiding} />
```

I notice the original `onBind` is call after binding, then use its past tense, `bound`, is more semantic.